### PR TITLE
Ignore unscheduled transaction results

### PR DESCRIPTION
**Pull request by peterschwarz**
When an transaction has been removed from the schedule, but was being
processed by a transaction processor, the scheduler will ignore any
results provided.

Includes a pair of unit tests for both schedulers to verify this
behavior.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

### CODE REVIEW

File: validator/sawtooth_validator/execution/scheduler_parallel.py [hunk 1]
```diff
@@ -649,6 +649,12 @@ def set_transaction_execution_result(
             if txn_signature not in self._scheduled:
                 raise SchedulerError(
                     "transaction not scheduled: {}".format(txn_signature))
+
+            if txn_signature not in self._batches_by_txn_id:
+                LOGGER.debug('Received result for %s, but was unscheduled',
```
Comment (nick-drozd) [inline validator/sawtooth_validator/execution/scheduler_parallel.py]: This feels like it will clutter the logs.

Comment (peterschwarz) [reply validator/sawtooth_validator/execution/scheduler_parallel.py]: I was using the same logging strategy from the serial scheduler.  I could remove it from here and the serial scheduler.

File: validator/tests/test_scheduler/tests.py [hunk 2]
```diff
@@ -428,6 +428,101 @@ def _test_wildcarded_unschedule(self, scheduler, context_manager):
...
+        self._test_unschedule(
+            scheduler=scheduler,
+            context_manager=context_manager)
+
+    def _test_unschedule(self, scheduler, context_manager):
+        """Tests that each of the top level methods on scheduler can be
+        called after unscheduling.
+
+        Notes:
+
```
Comment (agunde406) [inline validator/tests/test_scheduler/tests.py]: Missing Notes

Comment (peterschwarz) [reply validator/tests/test_scheduler/tests.py]: Notes added.

File: validator/sawtooth_validator/execution/scheduler_parallel.py (hunk 1 above)
Comment (nick-drozd) [reply validator/sawtooth_validator/execution/scheduler_parallel.py]: They could also be replaced with metrics.

**Pull request merged**